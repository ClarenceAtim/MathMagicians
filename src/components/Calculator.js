import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './Calculator.css';
import calculate from '../logic/calculate';

function Button({ value, className, onClick }) {
  return (
    <button type="button" className={className} onClick={onClick}>
      {value}
    </button>
  );
}

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  className: '',
};

function Calculator() {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });
  function handleButtonClick(buttonName) {
    const newData = calculate(calculatorData, buttonName);
    setCalculatorData(newData);
  }

  return (
    <div className="main">
      <h2>L3t$ D0 s0m3 M@th$</h2>
      <div className="cover">
        <div className="calculator">
          <div className="calculator__wrap">
            <div className="calculator__display">
              {calculatorData.next || calculatorData.total || '0'}
            </div>
            <div className="calculator__functions">
              <Button value="AC" onClick={() => handleButtonClick('AC')} />
              <Button value="+/-" onClick={() => handleButtonClick('+/-')} />
              <Button value="%" onClick={() => handleButtonClick('%')} />
              <Button value="รท" className="btn--opaque" onClick={() => handleButtonClick('รท')} />
              <Button value="7" onClick={() => handleButtonClick('7')} />
              <Button value="8" onClick={() => handleButtonClick('8')} />
              <Button value="9" onClick={() => handleButtonClick('9')} />
              <Button value="x" className="btn--opaque" onClick={() => handleButtonClick('x')} />
              <Button value="4" onClick={() => handleButtonClick('4')} />
              <Button value="5" onClick={() => handleButtonClick('5')} />
              <Button value="6" onClick={() => handleButtonClick('6')} />
              <Button value="-" className="btn--opaque" onClick={() => handleButtonClick('-')} />
              <Button value="1" onClick={() => handleButtonClick('1')} />
              <Button value="2" onClick={() => handleButtonClick('2')} />
              <Button value="3" onClick={() => handleButtonClick('3')} />
              <Button value="+" className="btn--opaque" onClick={() => handleButtonClick('+')} />
              <Button value="0" className="btn--large" onClick={() => handleButtonClick('0')} />
              <Button value="." onClick={() => handleButtonClick('.')} />
              <Button value="=" className="btn--opaque" onClick={() => handleButtonClick('=')} />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
