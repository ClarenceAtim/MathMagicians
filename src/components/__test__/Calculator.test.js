import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Calculator', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
