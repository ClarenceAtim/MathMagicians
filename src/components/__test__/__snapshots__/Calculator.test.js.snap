// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator matches the snapshot 1`] = `
<div>
  <div
    class="main"
  >
    <h2>
      L3t$ D0 s0m3 M@th$
    </h2>
    <div
      class="cover"
    >
      <div
        class="calculator"
      >
        <div
          class="calculator__wrap"
        >
          <div
            class="calculator__display"
          >
            0
          </div>
          <div
            class="calculator__functions"
          >
            <button
              class=""
              type="button"
            >
              AC
            </button>
            <button
              class=""
              type="button"
            >
              +/-
            </button>
            <button
              class=""
              type="button"
            >
              %
            </button>
            <button
              class="btn--opaque"
              type="button"
            >
              รท
            </button>
            <button
              class=""
              type="button"
            >
              7
            </button>
            <button
              class=""
              type="button"
            >
              8
            </button>
            <button
              class=""
              type="button"
            >
              9
            </button>
            <button
              class="btn--opaque"
              type="button"
            >
              x
            </button>
            <button
              class=""
              type="button"
            >
              4
            </button>
            <button
              class=""
              type="button"
            >
              5
            </button>
            <button
              class=""
              type="button"
            >
              6
            </button>
            <button
              class="btn--opaque"
              type="button"
            >
              -
            </button>
            <button
              class=""
              type="button"
            >
              1
            </button>
            <button
              class=""
              type="button"
            >
              2
            </button>
            <button
              class=""
              type="button"
            >
              3
            </button>
            <button
              class="btn--opaque"
              type="button"
            >
              +
            </button>
            <button
              class="btn--large"
              type="button"
            >
              0
            </button>
            <button
              class=""
              type="button"
            >
              .
            </button>
            <button
              class="btn--opaque"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
